NumLib/ODESolver/ODETypes.h

```diff
@@ -0,0 +1,212 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef NUMLIB_ODETYPES_H
+#define NUMLIB_ODETYPES_H
+
+// TODO: Move that file somewhere else.
+//       Actually it would be better to put all of this into the MathLib namespace
+//       in the end, because all matrices, vectors and linear solvers reside there.
+//       That will be done during some subsequent cleanup.
+
+#include <initializer_list>
+#include <cassert>
+
+
+// Dense Eigen matrix/vector //////////////////////////////////////////
+// always enabled
+
+#include <Eigen/LU>
+
+namespace NumLib
+{
+
+inline void setMatrix(Eigen::MatrixXd& m,
+                      Eigen::MatrixXd::Index const rows, Eigen::MatrixXd::Index const cols,
+                      std::initializer_list<double> values)
+{
+    using IndexType = Eigen::MatrixXd::Index;
```

Comment:
This index type does not correspond to the MathLib::EigenMatrix index type.
Also see a note in [EigenVector.h](https://github.com/ufz/ogs/blob/master/MathLib/LinAlg/Eigen/EigenVector.h#L34).

In general the matrix and vector types (and their indices) are in inconsistent state. I think the `MatrixTraits` shall be defined only for MathLib::XXXMatrix/Vector types.